Cleaned up CacheActionsRunnerConfig

tom-pollak opened the pull request:
# Description

Before `CacheActivationConfig` had a inconsistent config file for some interopability with `LanguageModelSAERunnerConfig`. It was kind of unclear which parameters were necessary vs redundant, and just was fairly unclear.

Simplified to the required arguments:

- `hf_dataset_path`: Tokenized or untokenized dataset
- `total_training_tokens`
- `model_name`
- `model_batch_size`
- `hook_name`
- `final_hook_layer`
- `d_in`

I think this scheme captures everything you need when attempting to cache activations and makes it a lot easier to reason about.

Optional:

```
activation_save_path # defaults to "activations/{dataset}/{model}/{hook_name}
shuffle=True
prepend_bos=True
streaming=True
seqpos_slice
buffer_size_gb=2 # Size of each buffer. Affects memory usage and saving freq
device="cuda" or "cpu"
dtype="float32"
autocast_lm=False
compile_llm=True
hf_repo_id # Push to hf
model_kwargs # `run_with_cache`
model_from_pretrained_kwargs
```

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have not rewritten tests relating to key interfaces which would affect backward compatibility

chanind commented on sae_lens/config.py:
```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
+    dataset_trust_remote_code: bool | None = None
+
+    # set in __post_init__
+    tokens_in_buffer: int = -1
```
nit: this would be better as `@property` functions rather than direct config params

chanind commented on sae_lens/config.py:
```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
-    new_cached_activations_path: Optional[str] = (
-        None  # Defaults to "activations/{dataset}/{model}/{full_hook_name}_{hook_head_index}"
-    )
+    hf_dataset_path: str
```
this looks like a breaking change. Do we need to change this name? This would require a major version bump and would be annoying for anyone currently using the library as they will need to change their configs

chanind commented on sae_lens/config.py:
```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
-    # SAE Parameters
-    d_in: int = 512
+    hook_name: str
+    final_hook_layer: int  # Layer of final hook. Same layer as hook_name
```
This also looks like a break change. Why is this renamed?

chanind commented on sae_lens/training/activations_store.py:
```diff
@@ -49,13 +49,77 @@ class ActivationsStore:
+        cls,
+        model: HookedRootModule,
+        cfg: CacheActivationsRunnerConfig,
+    ) -> "ActivationsStore":
```
Why is this different than `from_cache_activations` below? These both take a cache activations runner config and return a store

chanind commented on sae_lens/training/activations_store.py:
```diff
@@ -563,15 +627,17 @@ def get_buffer(
             )
 
-        refill_iterator = range(0, batch_size * n_batches_in_buffer, batch_size)
+        refill_iterator = range(0, total_size, batch_size)
```
👍

chanind commented on sae_lens/config.py:
```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
-    train_batch_size_tokens: int = 4096
-    normalize_activations: str = "none"  # should always be none for activation caching
-    seqpos_slice: tuple[int | None, ...] = (None,)
+    total_training_tokens: int
```
This also looks like a breaking change

chanind reviewed (commented):
It looks like this has some nice refactoring, but also a bunch of breaking changes. Do these params need to be renamed? It's nice if the params match what's in the other runner config as much as possible

tom-pollak commented:
sorry about that! renamed accordingly. what do you think about renaming `store_batch_size_prompts` to `model_batch_size`? I could keep `store_batch_size_prompts` as a deprecated param

Summary of changes to config:
```
- buffer_size_gb replaces n_batches_in_buffer

Removed:
- hook_head_index
- cached_activations_path
- is_dataset_tokenized 
- train_batch_size_tokens
- normalize_activations
- act_store_device
```

Technically shouldn't it still be breaking since we removed some parameters? or I could add a `**kwargs` to catch these

chanind commented:
I'd rather not use `**kwargs` since that would mean silently ignoring options. I'll just bump the major version, but it looks like for most users it should require fewer changes this time.
